tests/phpunit/Unit/Connection/CallbackConnectionProviderTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace SMW\Tests\Connection;
+
+use SMW\Connection\CallbackConnectionProvider;
+
+/**
+ * @covers \SMW\Connection\CallbackConnectionProvider
+ * @group semantic-mediawiki
+ *
+ * @license GNU GPL v2+
+ * @since 3.0
+ *
+ * @author mwjames
+ */
+class CallbackConnectionProviderTest extends \PHPUnit_Framework_TestCase {
+
+	public function testCanConstruct() {
+
+		$callback = function() {};
+
+		$this->assertInstanceOf(
+			CallbackConnectionProvider::class,
+			new CallbackConnectionProvider( $callback )
+		);
+	}
+
+	public function testGetConnection() {
```

Comment:
This is not a descriptive name. I suggest doing something like `testGetConnectionReturnsTheCallbacksReturnValue`.

Reply:
Feel free to changed it, it works for me.

Reply:
I'm not set on having this particular bit of code look one way or the other. What I am doing here with my time is giving you some advice, which you are of course free to ignore. Naming your test methods badly indeed "will work". Naming them more descriptive will work better, both for yourself, and to a bigger degree for other people. Not writing any tests also "will work". I recall giving advice to this one specific contributor several years ago that writing tests is a good idea, and at first getting a bunch of outraged "oh, so now I need to write tests for everything?!". I was happy to see this person changed their mind, wrote a lot of tests, and now encourages other people to do the same. If they had stayed at "it works for me", that'd have been a shame.

Reply:
Let not mix functional relevancy (which is to use a unit test to verify that what is expected can actually be asserted) with cosmetic preference which the later is about. We do agree that descriptive names may help others to understand the code in question but for a class with one single responsibility we don't have go in circles here and keep discussing a dead horse.

Reply:
So you are saying that in classes with a single responsibility good naming is not important as long as the classes do their job? And also that I am wasting my time even bringing this up?